### PR TITLE
Fixed over release of object on osx

### PR DESCRIPTION
**Pull request by mkareta**
Fixed over release of object that was autoreleased and then picked up by ARC as retained. 
Use method implementaion instead of NSInvocation as much faster way to call as well as much safer  and understandable by compoiler.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 6ed02427c0a32495e86fee19b52b9b4cb8990a09 (Job results: [3627](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3627/)) failed the code audit or documentation test.

File: osquery/tables/system/darwin/disk_encryption.mm [hunk 1]
```diff
@@ -234,48 +234,25 @@ void genFDEStatusForAPFS(Row& r) {
...
-    [inv setArgument:&targetVol atIndex:2];
-    [inv setArgument:&isEncryptedPtr atIndex:3];
-    [inv invokeWithTarget:apfs];
+    int (*function)(id, SEL, DADiskRef, char *) =
+      (int (*)(id, SEL, DADiskRef, char *))(methodIMP);
+    err = function(apfs, selector, targetVol, &isEncrypted);
   } @catch (NSException* exception) {
-    LOG(ERROR) << "NSInvocation threw for isEncryptedVolume:encrypted:";
+    LOG(ERROR)
+      << "isEncryptedVolume:encrypted: threw exception" << exception.name;
```
Comment (fmanco) [inline osquery/tables/system/darwin/disk_encryption.mm]: nit: missing white space.

File: osquery/tables/system/darwin/disk_encryption.mm [hunk 2]
```diff
@@ -286,44 +263,22 @@ void genFDEStatusForAPFS(Row& r) {
...
-    [inv setArgument:&targetVol atIndex:2];
-    [inv setArgument:&cryptoUsersPtr atIndex:3];
-    [inv invokeWithTarget:apfs];
+    int (*function)(id, SEL, DADiskRef, id *) =
+      (int (*)(id, SEL, DADiskRef, id __autoreleasing *))(methodIMP);
+    err = function(apfs, selector, targetVol, &cryptoUsers);
   } @catch (NSException* exception) {
-    LOG(ERROR) << "NSInvocation threw for cryptoUsersForVolume:users:";
+    LOG(ERROR)
+      << "cryptoUsersForVolume:users: threw exception" << exception.name;
```
Comment (fmanco) [inline osquery/tables/system/darwin/disk_encryption.mm]: nit: missing white space.

**Comment by fmanco** [comment]
ok to test to trigger darwin tests

**Comment by osqueryer** [comment]
:no_good: The commit 6ed02427c0a32495e86fee19b52b9b4cb8990a09 (Job results: [3633](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3633/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit 1628238ad8a21d12d7bf025d92c9615e8109c522 (Job results: [3643](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3643/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit c0a3f97508b0e7889c547ff91a3db52f1e7686b7 (Job results: [3644](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3644/)) failed the code audit or documentation test.

**Pull request merged**